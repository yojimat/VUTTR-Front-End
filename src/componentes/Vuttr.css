ul {
	display: inline-block;
	list-style-type: none;
}

a {
	display: none;
}

.App { text-align: center; }

@media (max-width: 600px) 
{
  .w-50{
  	width: 100%
  }
}
.pesquisaInput {
	background-image: url(../helpers/icones/lupaBossabox.svg);
	background-repeat: no-repeat;
	background-color: #F5F4F6;
	padding-left: 30px;
	border: 1px solid #EBEAED;
}

.linkTitle:hover .link {
	display: inline;
}

.container {
  display: inline-block;
  position: relative;
  padding-left: 25px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.container input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}

.checkmark {
  	position: absolute;
  	top: 0;
  	left: 0;
  	height: 20px;
  	width: 20px;
  	background-color: #F5F4F6;
  	border: 1px solid #DEDCE1;
}

.container:hover input ~ .checkmark {
  	background-color: #DEDCE1;
}

.container input:checked ~ .checkmark {
  	border: 1px solid #365DF0;
	background-color: #365DF0;
}
.checkmark:after {
  content: "";
  position: absolute;
  display: none;
}

/* Show the checkmark when checked */
.container input:checked ~ .checkmark:after {
  display: block;
}

/* Style the checkmark/indicator */
.container .checkmark:after {
  left: 6px;
  top: 2px;
  width: 5px;
  height: 10px;
  border: solid white;
  border-width: 0 3px 3px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}