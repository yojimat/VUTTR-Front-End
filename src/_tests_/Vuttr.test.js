import React from 'react';
import Vuttr from '../componentes/Vuttr';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

const abrirModal = jest.fn(),
	abrirModalDelete = jest.fn(),
	pesquisaFiltradasTools = jest.fn(),
	toggleTagFiltro = jest.fn(),
	defaultProps = {
		abrirModal,
		abrirModalDelete,
		toolsList: [],
		pesquisaFiltradasTools,
		toggleTagFiltro,
		ref: React.createRef()	
	}

const VuttrCriadorComponents = (props=defaultProps) =>{
	const VuttrDefaultComponent = mount(<><Vuttr {...props } /></>);
	return VuttrDefaultComponent;
}

describe(">>>Vuttr.js", () => {
	it("+Snapshot/Vuttr renderiza normalmente.", () => {  
	    const Vuttr = VuttrCriadorComponents();

	    expect(Vuttr).toMatchSnapshot();
	});

	it("+Verificando props toolsList", () => {
		const props = {
			toolsList: [{
	      		"id": 1,
		        "title": "Notion",
		        "link": "https://notion.so",
		        "description": "All in one tool to organize teams and ideas. Write, plan, collaborate, and get organized. ",
		        "tags": [
		        "organization",
		        "planning",
		        "collaboration",
		        "writing",
		        "calendar"
		     	]},
    			{
		      	"id": 2,
		      "title": "json-server",
		      "link": "https://github.com/typicode/json-server",
		      "description": "Fake REST API based on a json schema. Useful for mocking and creating APIs for front-end devs to consume in coding challenges.",
		      "tags": [
		        "api",
		        "json",
		        "schema",
		        "node",
		        "github",
		        "rest"
		      ]}
    		],
    		ref: React.createRef()
		},
			VuttrList = VuttrCriadorComponents(props).find("li");

		expect(VuttrList).toHaveLength(2);
		//Verifica lista tags, que com o modelo tem 11.
		expect(VuttrList.find("small")).toHaveLength(11);
		expect.assertions(2);
	});

	it("+Verificando evento ToogleDel", () => {
		const props = {
			toolsList: [{
	      		"id": 1,
		        "title": "Notion",
		        "link": "https://notion.so",
		        "description": "All in one tool to organize teams and ideas. Write, plan, collaborate, and get organized. ",
		        "tags": [
		        "organization",
		        "planning",
		        "collaboration",
		        "writing",
		        "calendar"
		     	]}
    		],
    		ref: React.createRef(),
			abrirModalDelete
		},
			VuttrEventDel = VuttrCriadorComponents(props).find("p + button");

		expect(VuttrEventDel).toHaveLength(1);
		VuttrEventDel.simulate('click', abrirModalDelete);
		expect(abrirModalDelete).toHaveBeenCalled();
		expect.assertions(2);
	});

	it("+Verificando evento ToogleAdd", () => {
		const VuttrEventAdd = VuttrCriadorComponents().find("button");

		expect(VuttrEventAdd).toHaveLength(1);
		VuttrEventAdd.simulate('click', abrirModal);
		expect(abrirModal).toHaveBeenCalled();
		expect.assertions(2);
	});

	it("+Verificando evento pesquisaFiltradasTools", () => {
		const VuttrEventPesquisaFiltrada = VuttrCriadorComponents().find("input[type='search']");

		expect(VuttrEventPesquisaFiltrada).toHaveLength(1);
		VuttrEventPesquisaFiltrada.simulate("change", pesquisaFiltradasTools);
		expect(pesquisaFiltradasTools).toHaveBeenCalled();
		expect.assertions(2);
	});

	it("+Verificando evento toggleTagFiltro", () => {
		const VuttrEventToggleTagFiltro = VuttrCriadorComponents().find("input[type='checkbox']");

		expect(VuttrEventToggleTagFiltro).toHaveLength(1);
		VuttrEventToggleTagFiltro.simulate("change", toggleTagFiltro);
		expect(toggleTagFiltro).toHaveBeenCalled();
		expect.assertions(2);
	});
});