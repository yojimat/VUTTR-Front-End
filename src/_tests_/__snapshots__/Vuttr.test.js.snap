// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Vuttr.js +Snapshot/Vuttr renderiza normalmente. 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <React.Fragment>
    <ForwardRef
      abrirModal={[MockFunction]}
      abrirModalDelete={[MockFunction]}
      pesquisaFiltradasTools={[MockFunction]}
      toggleTagFiltro={[MockFunction]}
      toolsList={Array []}
    />
  </React.Fragment>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "getWrappingComponentRenderer": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "abrirModal": [MockFunction],
      "abrirModalDelete": [MockFunction],
      "pesquisaFiltradasTools": [MockFunction],
      "toggleTagFiltro": [MockFunction],
      "toolsList": Array [],
    },
    "ref": Object {
      "current": null,
    },
    "rendered": Object {
      "instance": <div
        class="App"
      >
        <header>
          <h1
            class="ink mt4 mb0 i fw1"
          >
            VUTTR
          </h1>
          <h2
            class="ink mt2 mb4 ttu tracked"
          >
            Very Useful Tools to Remember
          </h2>
        </header>
        <section>
          <form>
            <input
              aria-label="Pesquisa"
              class="mb2 pesquisaInput br1"
              placeholder="Pesquise aqui…"
              type="search"
              value=""
            />
             
            <label
              class="lh-copy container ml3"
            >
              Procurar com #tags?
              <input
                type="checkbox"
              />
              <span
                class="checkmark br1"
              />
            </label>
          </form>
          <br />
          <button
            class="grow br2 ph3 pv2 mb2 dib bg-add azul"
            type="button"
          >
            <span
              aria-label="plus"
              role="img"
            >
              ➕
            </span>
             Adicionar Tool
          </button>
          <br />
          <article
            class="bg-branco-escuro w-50 center mb4 o-90 ba b--black-20 br2 pr4 shadowBox"
          >
            <div
              class="pa2"
            >
              <span>
                Lista vazia.
              </span>
            </div>
          </article>
        </section>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1
              className="ink mt4 mb0 i fw1"
            >
              VUTTR
            </h1>
            <h2
              className="ink mt2 mb4 ttu tracked"
            >
              Very Useful Tools to Remember
            </h2>
          </header>,
          <section>
            <form>
              <input
                aria-label="Pesquisa"
                className="mb2 pesquisaInput br1"
                onChange={[Function]}
                placeholder="Pesquise aqui…"
                type="search"
              />
               
              <label
                className="lh-copy container ml3"
              >
                Procurar com #tags?
                <input
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  className="checkmark br1"
                />
              </label>
            </form>
            <br />
            <button
              className="grow br2 ph3 pv2 mb2 dib bg-add azul"
              onClick={[MockFunction]}
              type="button"
            >
              <span
                aria-label="plus"
                role="img"
              >
                ➕
              </span>
               Adicionar Tool
            </button>
            <br />
            <article
              className="bg-branco-escuro w-50 center mb4 o-90 ba b--black-20 br2 pr4 shadowBox"
            >
              <div
                className="pa2"
              >
                <span>
                  Lista vazia.
                </span>
              </div>
            </article>
          </section>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <header>
            <h1
              class="ink mt4 mb0 i fw1"
            >
              VUTTR
            </h1>
            <h2
              class="ink mt2 mb4 ttu tracked"
            >
              Very Useful Tools to Remember
            </h2>
          </header>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="ink mt4 mb0 i fw1"
              >
                VUTTR
              </h1>,
              <h2
                className="ink mt2 mb4 ttu tracked"
              >
                Very Useful Tools to Remember
              </h2>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <h1
                class="ink mt4 mb0 i fw1"
              >
                VUTTR
              </h1>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "VUTTR",
                "className": "ink mt4 mb0 i fw1",
              },
              "ref": null,
              "rendered": Array [
                "VUTTR",
              ],
              "type": "h1",
            },
            Object {
              "instance": <h2
                class="ink mt2 mb4 ttu tracked"
              >
                Very Useful Tools to Remember
              </h2>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Very Useful Tools to Remember",
                "className": "ink mt2 mb4 ttu tracked",
              },
              "ref": null,
              "rendered": Array [
                "Very Useful Tools to Remember",
              ],
              "type": "h2",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": <section>
            <form>
              <input
                aria-label="Pesquisa"
                class="mb2 pesquisaInput br1"
                placeholder="Pesquise aqui…"
                type="search"
                value=""
              />
               
              <label
                class="lh-copy container ml3"
              >
                Procurar com #tags?
                <input
                  type="checkbox"
                />
                <span
                  class="checkmark br1"
                />
              </label>
            </form>
            <br />
            <button
              class="grow br2 ph3 pv2 mb2 dib bg-add azul"
              type="button"
            >
              <span
                aria-label="plus"
                role="img"
              >
                ➕
              </span>
               Adicionar Tool
            </button>
            <br />
            <article
              class="bg-branco-escuro w-50 center mb4 o-90 ba b--black-20 br2 pr4 shadowBox"
            >
              <div
                class="pa2"
              >
                <span>
                  Lista vazia.
                </span>
              </div>
            </article>
          </section>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <form>
                <input
                  aria-label="Pesquisa"
                  className="mb2 pesquisaInput br1"
                  onChange={[Function]}
                  placeholder="Pesquise aqui…"
                  type="search"
                />
                 
                <label
                  className="lh-copy container ml3"
                >
                  Procurar com #tags?
                  <input
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <span
                    className="checkmark br1"
                  />
                </label>
              </form>,
              <br />,
              <button
                className="grow br2 ph3 pv2 mb2 dib bg-add azul"
                onClick={[MockFunction]}
                type="button"
              >
                <span
                  aria-label="plus"
                  role="img"
                >
                  ➕
                </span>
                 Adicionar Tool
              </button>,
              <br />,
              <article
                className="bg-branco-escuro w-50 center mb4 o-90 ba b--black-20 br2 pr4 shadowBox"
              >
                <div
                  className="pa2"
                >
                  <span>
                    Lista vazia.
                  </span>
                </div>
              </article>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <form>
                <input
                  aria-label="Pesquisa"
                  class="mb2 pesquisaInput br1"
                  placeholder="Pesquise aqui…"
                  type="search"
                  value=""
                />
                 
                <label
                  class="lh-copy container ml3"
                >
                  Procurar com #tags?
                  <input
                    type="checkbox"
                  />
                  <span
                    class="checkmark br1"
                  />
                </label>
              </form>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <input
                    aria-label="Pesquisa"
                    className="mb2 pesquisaInput br1"
                    onChange={[Function]}
                    placeholder="Pesquise aqui…"
                    type="search"
                  />,
                  " ",
                  <label
                    className="lh-copy container ml3"
                  >
                    Procurar com #tags?
                    <input
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="checkmark br1"
                    />
                  </label>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <input
                    aria-label="Pesquisa"
                    class="mb2 pesquisaInput br1"
                    placeholder="Pesquise aqui…"
                    type="search"
                    value=""
                  />,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-label": "Pesquisa",
                    "className": "mb2 pesquisaInput br1",
                    "onChange": [Function],
                    "placeholder": "Pesquise aqui…",
                    "type": "search",
                    "value": undefined,
                  },
                  "ref": [Function],
                  "rendered": Array [
                    undefined,
                  ],
                  "type": "input",
                },
                " ",
                Object {
                  "instance": <label
                    class="lh-copy container ml3"
                  >
                    Procurar com #tags?
                    <input
                      type="checkbox"
                    />
                    <span
                      class="checkmark br1"
                    />
                  </label>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Procurar com #tags?",
                      <input
                        onChange={[Function]}
                        type="checkbox"
                      />,
                      <span
                        className="checkmark br1"
                      />,
                    ],
                    "className": "lh-copy container ml3",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Procurar com #tags?",
                    Object {
                      "instance": <input
                        type="checkbox"
                      />,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "onChange": [Function],
                        "type": "checkbox",
                      },
                      "ref": null,
                      "rendered": Array [
                        undefined,
                      ],
                      "type": "input",
                    },
                    Object {
                      "instance": <span
                        class="checkmark br1"
                      />,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "checkmark br1",
                      },
                      "ref": null,
                      "rendered": Array [
                        undefined,
                      ],
                      "type": "span",
                    },
                  ],
                  "type": "label",
                },
              ],
              "type": "form",
            },
            Object {
              "instance": <br />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "br",
            },
            Object {
              "instance": <button
                class="grow br2 ph3 pv2 mb2 dib bg-add azul"
                type="button"
              >
                <span
                  aria-label="plus"
                  role="img"
                >
                  ➕
                </span>
                 Adicionar Tool
              </button>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    aria-label="plus"
                    role="img"
                  >
                    ➕
                  </span>,
                  " Adicionar Tool",
                ],
                "className": "grow br2 ph3 pv2 mb2 dib bg-add azul",
                "onClick": [MockFunction],
                "type": "button",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <span
                    aria-label="plus"
                    role="img"
                  >
                    ➕
                  </span>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-label": "plus",
                    "children": "➕",
                    "role": "img",
                  },
                  "ref": null,
                  "rendered": Array [
                    "➕",
                  ],
                  "type": "span",
                },
                " Adicionar Tool",
              ],
              "type": "button",
            },
            Object {
              "instance": <br />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "br",
            },
            Object {
              "instance": <article
                class="bg-branco-escuro w-50 center mb4 o-90 ba b--black-20 br2 pr4 shadowBox"
              >
                <div
                  class="pa2"
                >
                  <span>
                    Lista vazia.
                  </span>
                </div>
              </article>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="pa2"
                >
                  <span>
                    Lista vazia.
                  </span>
                </div>,
                "className": "bg-branco-escuro w-50 center mb4 o-90 ba b--black-20 br2 pr4 shadowBox",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <div
                    class="pa2"
                  >
                    <span>
                      Lista vazia.
                    </span>
                  </div>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <span>
                      Lista vazia.
                    </span>,
                    "className": "pa2",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": <span>
                        Lista vazia.
                      </span>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Lista vazia.",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Lista vazia.",
                      ],
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "article",
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "abrirModal": [MockFunction],
        "abrirModalDelete": [MockFunction],
        "pesquisaFiltradasTools": [MockFunction],
        "toggleTagFiltro": [MockFunction],
        "toolsList": Array [],
      },
      "ref": Object {
        "current": null,
      },
      "rendered": Object {
        "instance": <div
          class="App"
        >
          <header>
            <h1
              class="ink mt4 mb0 i fw1"
            >
              VUTTR
            </h1>
            <h2
              class="ink mt2 mb4 ttu tracked"
            >
              Very Useful Tools to Remember
            </h2>
          </header>
          <section>
            <form>
              <input
                aria-label="Pesquisa"
                class="mb2 pesquisaInput br1"
                placeholder="Pesquise aqui…"
                type="search"
                value=""
              />
               
              <label
                class="lh-copy container ml3"
              >
                Procurar com #tags?
                <input
                  type="checkbox"
                />
                <span
                  class="checkmark br1"
                />
              </label>
            </form>
            <br />
            <button
              class="grow br2 ph3 pv2 mb2 dib bg-add azul"
              type="button"
            >
              <span
                aria-label="plus"
                role="img"
              >
                ➕
              </span>
               Adicionar Tool
            </button>
            <br />
            <article
              class="bg-branco-escuro w-50 center mb4 o-90 ba b--black-20 br2 pr4 shadowBox"
            >
              <div
                class="pa2"
              >
                <span>
                  Lista vazia.
                </span>
              </div>
            </article>
          </section>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header>
              <h1
                className="ink mt4 mb0 i fw1"
              >
                VUTTR
              </h1>
              <h2
                className="ink mt2 mb4 ttu tracked"
              >
                Very Useful Tools to Remember
              </h2>
            </header>,
            <section>
              <form>
                <input
                  aria-label="Pesquisa"
                  className="mb2 pesquisaInput br1"
                  onChange={[Function]}
                  placeholder="Pesquise aqui…"
                  type="search"
                />
                 
                <label
                  className="lh-copy container ml3"
                >
                  Procurar com #tags?
                  <input
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <span
                    className="checkmark br1"
                  />
                </label>
              </form>
              <br />
              <button
                className="grow br2 ph3 pv2 mb2 dib bg-add azul"
                onClick={[MockFunction]}
                type="button"
              >
                <span
                  aria-label="plus"
                  role="img"
                >
                  ➕
                </span>
                 Adicionar Tool
              </button>
              <br />
              <article
                className="bg-branco-escuro w-50 center mb4 o-90 ba b--black-20 br2 pr4 shadowBox"
              >
                <div
                  className="pa2"
                >
                  <span>
                    Lista vazia.
                  </span>
                </div>
              </article>
            </section>,
          ],
          "className": "App",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <header>
              <h1
                class="ink mt4 mb0 i fw1"
              >
                VUTTR
              </h1>
              <h2
                class="ink mt2 mb4 ttu tracked"
              >
                Very Useful Tools to Remember
              </h2>
            </header>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="ink mt4 mb0 i fw1"
                >
                  VUTTR
                </h1>,
                <h2
                  className="ink mt2 mb4 ttu tracked"
                >
                  Very Useful Tools to Remember
                </h2>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <h1
                  class="ink mt4 mb0 i fw1"
                >
                  VUTTR
                </h1>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "VUTTR",
                  "className": "ink mt4 mb0 i fw1",
                },
                "ref": null,
                "rendered": Array [
                  "VUTTR",
                ],
                "type": "h1",
              },
              Object {
                "instance": <h2
                  class="ink mt2 mb4 ttu tracked"
                >
                  Very Useful Tools to Remember
                </h2>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Very Useful Tools to Remember",
                  "className": "ink mt2 mb4 ttu tracked",
                },
                "ref": null,
                "rendered": Array [
                  "Very Useful Tools to Remember",
                ],
                "type": "h2",
              },
            ],
            "type": "header",
          },
          Object {
            "instance": <section>
              <form>
                <input
                  aria-label="Pesquisa"
                  class="mb2 pesquisaInput br1"
                  placeholder="Pesquise aqui…"
                  type="search"
                  value=""
                />
                 
                <label
                  class="lh-copy container ml3"
                >
                  Procurar com #tags?
                  <input
                    type="checkbox"
                  />
                  <span
                    class="checkmark br1"
                  />
                </label>
              </form>
              <br />
              <button
                class="grow br2 ph3 pv2 mb2 dib bg-add azul"
                type="button"
              >
                <span
                  aria-label="plus"
                  role="img"
                >
                  ➕
                </span>
                 Adicionar Tool
              </button>
              <br />
              <article
                class="bg-branco-escuro w-50 center mb4 o-90 ba b--black-20 br2 pr4 shadowBox"
              >
                <div
                  class="pa2"
                >
                  <span>
                    Lista vazia.
                  </span>
                </div>
              </article>
            </section>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <form>
                  <input
                    aria-label="Pesquisa"
                    className="mb2 pesquisaInput br1"
                    onChange={[Function]}
                    placeholder="Pesquise aqui…"
                    type="search"
                  />
                   
                  <label
                    className="lh-copy container ml3"
                  >
                    Procurar com #tags?
                    <input
                      onChange={[Function]}
                      type="checkbox"
                    />
                    <span
                      className="checkmark br1"
                    />
                  </label>
                </form>,
                <br />,
                <button
                  className="grow br2 ph3 pv2 mb2 dib bg-add azul"
                  onClick={[MockFunction]}
                  type="button"
                >
                  <span
                    aria-label="plus"
                    role="img"
                  >
                    ➕
                  </span>
                   Adicionar Tool
                </button>,
                <br />,
                <article
                  className="bg-branco-escuro w-50 center mb4 o-90 ba b--black-20 br2 pr4 shadowBox"
                >
                  <div
                    className="pa2"
                  >
                    <span>
                      Lista vazia.
                    </span>
                  </div>
                </article>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <form>
                  <input
                    aria-label="Pesquisa"
                    class="mb2 pesquisaInput br1"
                    placeholder="Pesquise aqui…"
                    type="search"
                    value=""
                  />
                   
                  <label
                    class="lh-copy container ml3"
                  >
                    Procurar com #tags?
                    <input
                      type="checkbox"
                    />
                    <span
                      class="checkmark br1"
                    />
                  </label>
                </form>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <input
                      aria-label="Pesquisa"
                      className="mb2 pesquisaInput br1"
                      onChange={[Function]}
                      placeholder="Pesquise aqui…"
                      type="search"
                    />,
                    " ",
                    <label
                      className="lh-copy container ml3"
                    >
                      Procurar com #tags?
                      <input
                        onChange={[Function]}
                        type="checkbox"
                      />
                      <span
                        className="checkmark br1"
                      />
                    </label>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <input
                      aria-label="Pesquisa"
                      class="mb2 pesquisaInput br1"
                      placeholder="Pesquise aqui…"
                      type="search"
                      value=""
                    />,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-label": "Pesquisa",
                      "className": "mb2 pesquisaInput br1",
                      "onChange": [Function],
                      "placeholder": "Pesquise aqui…",
                      "type": "search",
                      "value": undefined,
                    },
                    "ref": [Function],
                    "rendered": Array [
                      undefined,
                    ],
                    "type": "input",
                  },
                  " ",
                  Object {
                    "instance": <label
                      class="lh-copy container ml3"
                    >
                      Procurar com #tags?
                      <input
                        type="checkbox"
                      />
                      <span
                        class="checkmark br1"
                      />
                    </label>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Procurar com #tags?",
                        <input
                          onChange={[Function]}
                          type="checkbox"
                        />,
                        <span
                          className="checkmark br1"
                        />,
                      ],
                      "className": "lh-copy container ml3",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Procurar com #tags?",
                      Object {
                        "instance": <input
                          type="checkbox"
                        />,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "onChange": [Function],
                          "type": "checkbox",
                        },
                        "ref": null,
                        "rendered": Array [
                          undefined,
                        ],
                        "type": "input",
                      },
                      Object {
                        "instance": <span
                          class="checkmark br1"
                        />,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "checkmark br1",
                        },
                        "ref": null,
                        "rendered": Array [
                          undefined,
                        ],
                        "type": "span",
                      },
                    ],
                    "type": "label",
                  },
                ],
                "type": "form",
              },
              Object {
                "instance": <br />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "br",
              },
              Object {
                "instance": <button
                  class="grow br2 ph3 pv2 mb2 dib bg-add azul"
                  type="button"
                >
                  <span
                    aria-label="plus"
                    role="img"
                  >
                    ➕
                  </span>
                   Adicionar Tool
                </button>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span
                      aria-label="plus"
                      role="img"
                    >
                      ➕
                    </span>,
                    " Adicionar Tool",
                  ],
                  "className": "grow br2 ph3 pv2 mb2 dib bg-add azul",
                  "onClick": [MockFunction],
                  "type": "button",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <span
                      aria-label="plus"
                      role="img"
                    >
                      ➕
                    </span>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-label": "plus",
                      "children": "➕",
                      "role": "img",
                    },
                    "ref": null,
                    "rendered": Array [
                      "➕",
                    ],
                    "type": "span",
                  },
                  " Adicionar Tool",
                ],
                "type": "button",
              },
              Object {
                "instance": <br />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "br",
              },
              Object {
                "instance": <article
                  class="bg-branco-escuro w-50 center mb4 o-90 ba b--black-20 br2 pr4 shadowBox"
                >
                  <div
                    class="pa2"
                  >
                    <span>
                      Lista vazia.
                    </span>
                  </div>
                </article>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="pa2"
                  >
                    <span>
                      Lista vazia.
                    </span>
                  </div>,
                  "className": "bg-branco-escuro w-50 center mb4 o-90 ba b--black-20 br2 pr4 shadowBox",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <div
                      class="pa2"
                    >
                      <span>
                        Lista vazia.
                      </span>
                    </div>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <span>
                        Lista vazia.
                      </span>,
                      "className": "pa2",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": <span>
                          Lista vazia.
                        </span>,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Lista vazia.",
                        },
                        "ref": null,
                        "rendered": Array [
                          "Lista vazia.",
                        ],
                        "type": "span",
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "article",
              },
            ],
            "type": "section",
          },
        ],
        "type": "div",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
